src/node/CarplayWS.ts

```diff
@@ -0,0 +1,104 @@
+import { webusb } from 'usb'
+import NodeMicrophone from './NodeMicrophone'
+import EventEmitter from 'events'
+import {
+  AudioData,
+  Key,
+  MediaData,
+  Message,
+  Plugged,
+  SendAudio,
+  SendCarPlay,
+  SendTouch,
+  Unplugged,
+  VideoData,
+  DongleDriver,
+  DongleConfig,
+  DEFAULT_CONFIG,
+} from '../modules'
+
+export default class CarplayWS extends EventEmitter{
+  private _pairTimeout: NodeJS.Timeout | null = null
+  private _plugged = false
+  private _dongleDriver: DongleDriver
+
+  constructor(config: DongleConfig = DEFAULT_CONFIG, port = 5005) {
+    super()
+    const mic = new NodeMicrophone()
+    const driver = new DongleDriver()
+    mic.on('data', data => {
+      driver.send(new SendAudio(data))
+    })
+    driver.on('ready', async () => {
+      const { open, send } = driver
+
+      await open(config)
+
+      this._pairTimeout = setTimeout(() => {
+        console.debug('no device, sending pair')
+        send(new SendCarPlay('wifiPair'))
+      }, 15000)
+    })
+    driver.on('message', (message: Message) => {
+      if (message instanceof Plugged) {
+        if (this._pairTimeout) {
+          clearTimeout(this._pairTimeout)
+          this._pairTimeout = null
+        }
+        this._plugged = true
+        this.emitPlugged()
+      } else if (message instanceof Unplugged) {
+        this._plugged = false
+        this.emit('quit')
+      } else if (message instanceof VideoData) {
+        this.emit('carplay', message)
+      } else if (message instanceof AudioData) {
+        this.emit('audio', message)
+      } else if (message instanceof MediaData) {
+        this.emit('media', message)
+      }
```

Comment:
These all now emit the message from CarplayWS

Reply:
```JavaScript
  carplay.on('carplay', (message) => {
    mainWindow?.webContents.send('videoData', message);
  });

  carplay.on('audio', (message) => {
    mainWindow?.webContents.send('audioData', message);
  });
```

Example of how I have currently implemented it for electron

Reply:
This looks great!

You might want to also emit carplay messages like CarplayWeb does:

`else if (message instanceof CarPlay) {
        this.onmessage?.({ type: 'carplay', message })
      }`
      
     That way - you can get a message if users click the Car icon in the carplay interface and switch to you own react app